pkg/infra/iac2/templates_compiler.go

```diff
@@ -1,26 +1,32 @@
 package iac2
 
 import (
+	"bytes"
 	"embed"
 	_ "embed"
+	"encoding/json"
 	"fmt"
-	"github.com/klothoplatform/klotho/pkg/core"
-	"github.com/klothoplatform/klotho/pkg/graph"
-	"github.com/klothoplatform/klotho/pkg/multierr"
-	"github.com/pkg/errors"
-	"go.uber.org/zap"
 	"io"
 	"io/fs"
 	"reflect"
 	"regexp"
 	"sort"
 	"strings"
+
+	"github.com/klothoplatform/klotho/pkg/core"
+	"github.com/klothoplatform/klotho/pkg/graph"
+	"github.com/klothoplatform/klotho/pkg/lang/javascript"
+	"github.com/klothoplatform/klotho/pkg/multierr"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
 )
 
 type (
 	templatesCompiler struct {
 		// templates is the fs.FS where we read all of our `<struct>/factory.ts` files
 		templates fs.FS
+		// packageFiles is the fs.FS where we read all of our `<struct>/package.json` files
+		packageFiles fs.FS
```

Comment:
it wouldnt be different i wouldnt think, but i wasnt sure so i split it out. We can definitely keep them as one if we want, but my other thought is for standardizing the templates vs the project files maybe it would be useful. Not sure though

Reply:
I'd prefer to keep them as one for now:

1. Within Go code it keeps things simpler and with less chance of divergence (where we set one field but forget to set the other)
2. Within the file structure itself, I think we always want each `factory.ts` to sit next to its `package.json` so that we can do an `npm install && npx tsc && npx depcheck` on it.